src/ts_catalog/continuous_agg.c

```diff
@@ -1184,6 +1185,13 @@ drop_continuous_agg(FormData_continuous_agg *cadata, bool drop_user_view)
 
 	if (OidIsValid(mat_hypertable.objectId))
 	{
+#if PG14_GE
+		OsmCallbacks *callbacks = ts_get_osm_callbacks();
+
+		if (callbacks)
+			callbacks->hypertable_drop_hook(&mat_hypertable);
```

Comment:
why do we need a different hook type? Can't we just use the drop_by_name hook.
ts_hypertable_get_by_id() will fetch the hypertable info and we can get the schema_name + table_name from there.

Reply:
Good point, thanks!

Reply:
Removed one redundant hook